sim/humanoid_gym/play_h5.py

```diff
@@ -0,0 +1,213 @@
+import argparse
+import logging
+import os
+from datetime import datetime
+
+import cv2
+import h5py
+import numpy as np
+from isaacgym import gymapi
+from tqdm import tqdm
+
+from sim.logging import configure_logging
+
+logger = logging.getLogger(__name__)
+import copy
+
+from sim.env import run_dir
+from sim.humanoid_gym.envs import *  # noqa: F403
+
+
+def export_policy_as_jit(actor_critic, path):
+    os.makedirs(path, exist_ok=True)
+    path = os.path.join(path, "policy_1.pt")
+    model = copy.deepcopy(actor_critic.actor).to("cpu")
+    traced_script_module = torch.jit.script(model)
+    traced_script_module.save(path)
+
+
+def play(args: argparse.Namespace) -> None:
+    configure_logging()
+
+    logger.info("Configuring environment and training settings...")
+    env_cfg, train_cfg = task_registry.get_cfgs(name=args.task)
+    # override some parameters for testing
+    env_cfg.env.num_envs = min(env_cfg.env.num_envs, 1)
+    env_cfg.sim.max_gpu_contact_pairs = 2**10
+    env_cfg.terrain.mesh_type = "plane"
+    env_cfg.terrain.num_rows = 5
+    env_cfg.terrain.num_cols = 5
+    env_cfg.terrain.curriculum = False
+    env_cfg.terrain.max_init_terrain_level = 5
+    env_cfg.noise.add_noise = True
+    env_cfg.domain_rand.push_robots = True
+    env_cfg.domain_rand.joint_angle_noise = 0.0
+    env_cfg.noise.curriculum = False
+    env_cfg.noise.noise_level = 0.5
+
+    train_cfg.seed = 123145
+    logger.info("train_cfg.runner_class_name: %s", train_cfg.runner_class_name)
+
+    # prepare environment
+    env, _ = task_registry.make_env(name=args.task, args=args, env_cfg=env_cfg)
+    env.set_camera(env_cfg.viewer.pos, env_cfg.viewer.lookat)
+
+    obs = env.get_observations()
+    # breakpoint()e
+
+    # load policy
+    train_cfg.runner.resume = True
+    ppo_runner, train_cfg = task_registry.make_alg_runner(env=env, name=args.task, args=args, train_cfg=train_cfg)
+    policy = ppo_runner.get_inference_policy(device=env.device)
+
+    # Export policy if needed
+    EXPORT_POLICY = True
+    if EXPORT_POLICY:
+        path = os.path.join(".")
+        export_policy_as_jit(ppo_runner.alg.actor_critic, path)
+        print("Exported policy as jit script to: ", path)
+
+    # Prepare for logging
+    env_logger = Logger(env.dt)
+    robot_index = 0
+    joint_index = 1
+    stop_state_log = 1000
+
+    # Initialize HDF5 file
+    now = datetime.now()
+    now = now.strftime("%Y-%m-%d_%H-%M-%S")
+    h5_file = h5py.File("data" + now + ".h5", "w")
```

Comment:
@WT-MM - can you have a look

Reply:
https://github.com/kscalelabs/sim/pull/50#discussion_r1712163496